### DISCUSSION TITLE
Go Frontend: What is key.json?

### DISCUSSION THREAD

**Discussion by mbalug7** [zitadel/zitadel#7310] [Q&A] [answered]
I followed the instructions on creating a frontend web service (https://zitadel.com/docs/examples/login/go), and I'm not sure from where I should download the `key.json` file? Can you help me with that?

https://zitadel.com/docs/examples/login/go#start-your-application
`key: The path to the downloaded key.json`

**Comment by hifabienne** [comment]
Hei @mbalug7 
When you go to the zitadel console and to your service account.
There is a sidemenu with keys.
You can generate a new key there and download it

**Comment by mbalug7** [comment]
I tried to run my application with the following command:
`go run main.go --domain auth.ourdomain.com --key XKv2Lqd7YAq13NUZVUWZEWZeruqyzViM --clientID 251705954687582568@our_org_name --redirectURI http://localhost:8089/auth/callback`

The key I used is the key from Zitadel docs (you can find it here: https://zitadel.com/docs/examples/login/go#start-your-application)  `--key XKv2Lqd7YAq13NUZVUWZEWZeruqyzViM` - it is not a file, just a string, probably some AES key.

My app works perfectly fine with that key.

It is still not clear to me what is the purpose of that key, especially if PKCE flow is used, and as far as I understand, I don't have to create service account which has nothing to do with my web app and user login flow. 

From what I wrote, is there anything I don't understand or am I missing some detail?

**Reply by hifabienne** [reply]
@livio-a can you help here?

**Reply by livio-a** [reply, marked as answer]
Looks like we missed something when we updated the guide to the github.com/zitadel/zitadel-go/v3 package.

As you noticed, the key is a simple string. It's used to encrypt cookies, e.g. securing the `state` in the PKCE flow or the session cookie after successful authentication.

We'll update the docs and example to clarify that.

**Reply by mbalug7** [reply]
Great, thanks for the answer.
